src/box/engine.h

```diff
@@ -88,6 +88,20 @@ struct engine_memory_stat {
 	size_t tx;
 };
 
+/**
+ * Cursor used during checkpoint initial join.
+ */
+struct engine_checkpoint_cursor {
+	/** Flag, which indicates, whether checkpoint join should be done. */
+	bool is_checkpoint_join;
```

Comment:
- I think we could name the structure simply `checkpoint_cursor`.
- I don't think we really need a flag here - instead we could pass `NULL` for `cursor` to `engine_join`.

Reply:
Both points were fixed